### PR TITLE
[Email] Fix email related issues 

### PR DESCRIPTION
**Pull request by DingmaomaoBJTU**
## Description
1- Fix issue that delete email not working in Skill mode. (Root cause: mode arg didn't get passed to delete flow in skill mode.)
2- Fix email input not working in CN (Root cause: entity failed to recognize)
3- Fix some localization related bug
4- Add forward/reply flow support while show emails
5- Add support to query all emails
6- Add state clear control

## Related Issue
Cannot delete email on skill mode #347

## Testing Steps
Use VA main bot

User: Show my emails
Bot: (list of emails)
User: Delete the first one
Bot: Are you sure to delete?

Use VA main bot CN
1. User: 退出
2. Bot: 你已成功退出

Use VA main bot CN
1. User: 发一封邮件
2. Bot: 发给谁?
3. User: test@gmail.com
4. Bot: 好的, 邮件将发送给test@gmail.com.

## Checklist
- [1] I have commented my code, particularly in hard-to-understand areas
- [1] I have added or updated the appropriate unit tests
- [1] I have updated related documentation

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "botFilePath": ".\\YOUR_BOT_PATH.bot",
+  "botFilePath": ".\\CAFeichCN.bot",
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json]: revert this change

File: solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json [hunk 2]
```diff
@@ -37,8 +37,7 @@
       "assembly": "EmailSkill.EmailSkill, EmailSkill, Version=1.0.0.0, Culture=neutral",
       "dispatchIntent": "l_Email",
       "supportedProviders": [
-        "Azure Active Directory v2",
-        "Google"
+        "Azure Active Directory v2"
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json]: i guess we should remove Google until gmail support change is in. in fact it should be part of the gmail support change

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Resources/CommonStrings.zh.resx [hunk 3]
```diff
@@ -225,4 +225,10 @@
   <data name="Yesterday" xml:space="preserve">
     <value>昨天</value>
   </data>
+  <data name="All" xml:space="preserve">
+    <value>全部</value>
+  </data>
+  <data name="High" xml:space="preserve">
+    <value>特別</value>
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Resources/CommonStrings.zh.resx]: is this right?

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillDialog.cs [hunk 4]
```diff
@@ -158,11 +158,6 @@ private async Task InitializeSkill(DialogContext dc)
                     // set up skill turn error handling
                     OnTurnError = async (context, exception) =>
                     {
-                        await context.SendActivityAsync(context.Activity.CreateReply($"Sorry, something went wrong trying to communicate with the skill. Please try again."));
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillDialog.cs]: why removing this?

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillDialog.cs (hunk 4 above)
Comment (DingmaomaoBJTU) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillDialog.cs]: Ah, because error message already send by skills, so delete this. And this one also not localized~

Comment (lzc850612) [reply solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillDialog.cs]: I have made a change to localize this. also, this is needed because we need a global error handling because there's gonna be unhandled exceptions in skills.

**Comment by lauren-mills** [comment]
Just to clarify, this catches any unhandled errors in the skills. For example, if the email skill has issues connecting to the msgraph, catches a 400, responds to the user, but doesn’t actually “throw” that exception back to VA, this message won’t be sent. It’s only when there are exceptions that we don’t catch in the skill that this is called.

**Comment by DingmaomaoBJTU** [comment]
@lauren-mills Got it, will keep this~ Thanks!

File: solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json (hunk 2 above)
Comment (DingmaomaoBJTU) [reply solutions/Virtual-Assistant/src/csharp/assistant/appsettings.json]: Sure, thanks!

**Review by lzc850612 — APPROVED** [review]
